### ISSUE TITLE
Remove compilation warnings that can be removed

### ISSUE THREAD

**Issue by ericbottard** [reactor/reactor-core#2239]
This is a newly introduced compilation warning:
```
/Users/ericbottard/Documents/Projects/reactor-umbrella/reactor-core/reactor-core/src/main/java/reactor/core/publisher/FluxConcatMapNoPrefetch.java:173: warning: [unchecked] unchecked cast
                                        Callable<R> callable = (Callable<R>) p;
                                                                             ^
  required: Callable<R>
  found:    Publisher<CAP#1>
  where R is a type-variable:
    R extends Object declared in class FluxConcatMapNoPrefetchSubscriber
  where CAP#1 is a fresh type-variable:
    CAP#1 extends R from capture of ? extends R
```

**Comment by bsideup** [comment]
@ericbottard could you please configure the build to fail on warnings?

**Comment by ericbottard** [comment]
I'll look into it, not sure we can distinguish between those warnings and the remaining ones though. Created #2240

**Comment by ericbottard** [comment]
Changing the title of this issue to cover the 150+ warnings that we have (and that I previously didn't see because of a failing test locally ;-) )

**Comment by ericbottard** [comment]
Closed by 56e5c0dd
